src/main.py

```
@@ -1,2 +1,15 @@
-def substructure_search(mols, mol):
-    pass
\ No newline at end of file
+from rdkit import Chem
+
+
+def substructure_search(mols, mol): 
+    result = []
+    try:
+        for test_mol in mols:
+            if Chem.MolFromSmiles(test_mol).HasSubstructMatch(Chem.MolFromSmiles(mol)):
+                result.append(test_mol)
+        return result
+    except:
+        print("Something went wrong")
```

It is a bad practice to suppress errors - this can lead to bigger failures in the future. It would be better to remove exception handling here.